community/kernel/src/main/java/org/neo4j/kernel/impl/factory/DataSourceModule.java

```diff
@@ -421,6 +428,37 @@ private Procedures setupProcedures( PlatformModule platform, EditionModule editi
         return procedures;
     }
 
+    private ComponentRegistry.Provider<TerminationGuard> procedureGuard( final Dependencies dependencies )
+    {
+        Guard guard = dependencies.resolveDependency( Guard.class );
+        return (ctx) ->
+        {
+            KernelTransaction ktx = ctx.get( KERNEL_TRANSACTION );
+            return (TerminationGuard) () ->
```

Comment:
please extract this class into separate class and put it into some procedure related package, most probably where `TerminationGuard` live